network: canonicalization between IP and Ip

After adding the following configuration to r**esource-manager.hcl**, I ran make import under **tools/importer-rest-api-specs**. Then I got the following error when running **dotnet format whitespace --verbosity quiet ../../data/Pandora.sln**

service "network" {
  name      = "Network"
  available = ["2021-08-01"]
}

```
2022/05/11 14:13:13 ✅ Service "Network" - Api Version "2021-08-01"
# note we're intentionally only using whitespace here since the others try
# to fix design issues, which are actually issues in the API Data
dotnet format whitespace --verbosity quiet ../../data/Pandora.sln
Unhandled exception: System.AggregateException: One or more errors occurred. (Changes must be within bounds of SourceText (Parameter 'changes'))
 ---> System.ArgumentException: Changes must be within bounds of SourceText (Parameter 'changes')
   at Microsoft.CodeAnalysis.Text.SourceText.WithChanges(IEnumerable`1 changes)
   at Microsoft.CodeAnalysis.LinkedFileDiffMergingSession.MergeLinkedDocumentGroupAsync(IEnumerable`1 allLinkedDocuments, IEnumerable`1 linkedDocumentGroup, LinkedFileDiffMergingSessionInfo sessionInfo, IMergeConflictHandler mergeConflictHandler, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.LinkedFileDiffMergingSession.MergeDiffsAsync(IMergeConflictHandler mergeConflictHandler, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Solution.WithMergedLinkedFileChangesAsync(Solution oldSolution, Nullable`1 solutionChanges, IMergeConflictHandler mergeConflictHandler, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.TryApplyChanges(Solution newSolution)
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.FormatWorkspaceAsync(FormatOptions formatOptions, ILogger logger, CancellationToken cancellationToken, String binaryLogPath)
   at Microsoft.CodeAnalysis.Tools.FormatCommandCommon.FormatAsync(FormatOptions formatOptions, ILogger`1 logger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.Commands.FormatWhitespaceCommand.FormatWhitespaceHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass27_0.<<UseVersionOption>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__24_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseDebugDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseExceptionHandler>b__0>d.MoveNext()
make: *** [GNUmakefile:13: import] Error 1
```

--- a/tools/importer-rest-api-specs/components/parser/cleanup/cleanup.go
+++ b/tools/importer-rest-api-specs/components/parser/cleanup/cleanup.go
@@ -53,6 +53,7 @@ func RemoveInvalidCharacters(input string, titleCaseSegments bool) string {
 func NormalizeName(input string) string {
 	output := input
 	output = wordifyFirstCharacter(output)
+	output = NormalizeCanonicalisation(output)
 	output = RemoveInvalidCharacters(output, true)
 	output = NormalizeSegment(output, false)
 	output = strings.Title(output)
@@ -516,4 +517,28 @@ func wordifyFirstCharacter(input string) string {
 		return output
 	}
 	return input
+}
+
+func NormalizeCanonicalisation(input string) string {
+	output := input
+
+	// Some IP references have 'ip' as a prefix
+	if strings.HasPrefix(output, "ip") {
+		output = strings.Replace(output, "ip", "IP", 1)
+	}
+
+	if strings.EqualFold(output, "Publicipaddress") {
+		// This is an explicit force for broken data in `Network`
+		output = "PublicIPAddress"
+	}
+
+	if strings.EqualFold(output, "IPconfiguration") {
+		// This is an explicit force for broken data in `Network`
+		output = "IPConfiguration"
+	}
+
+	// intentionally case-sensitive
+	output = strings.ReplaceAll(output, "Ip", "IP")
+
+	return output
 }

--- a/tools/importer-rest-api-specs/components/parser/resourceids/generate_names.go
+++ b/tools/importer-rest-api-specs/components/parser/resourceids/generate_names.go
@@ -114,7 +114,7 @@ func generateNamesForResourceIds(input []models.ParsedResourceId, log hclog.Logg
 	// now we have unique ID's, we should go through and suffix `Id` onto the end of each of them
 	outputNamesToUris := make(map[string]models.ParsedResourceId)
 	for k, v := range candidateNamesToUris {
-		key := fmt.Sprintf("%sId", k)
+		key := fmt.Sprintf("%sId", cleanup.NormalizeName(k))
 		outputNamesToUris[key] = v
 	}
 

